Feature/odaa 154 enter frequency on flow

lats1 opened the pull request:
https://jira.itkdev.dk/browse/ODAA-154

lats1 committed:
ODAA-154: Add frequency field to dataflow form

lats1 committed:
ODAA-154: Adding cron expression bundle

lats1 committed:
ODAA-154: Replacing frequency with cron like shedule

rimi-itk commented on src/Command/DataFlow/DataFlowProduceJobsCommand.php:
```diff
@@ -96,25 +100,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
-            if ($dataFlow->getFrequency() < $seconds) {
-                return true;
+            if (null !== $dataFlow->getSchedule()) {
+                $cron = CronExpression::factory(\strval($dataFlow->getSchedule()));
```
Is `strval` needed here? What is the type of `$dataFlow->getSchedule()` (if not null)?

lats1 replied:
It actually a CronExpression object, so I don't need to use strval, but my editor didn't recognize it at first.

rimi-itk commented on src/Entity/DataFlow.php:
```diff
@@ -307,4 +295,16 @@ public function removeCollaborator(User $collaborator): self
         return $this;
     }
+
+    public function getSchedule()
```
Can we type hint here?

lats1 replied:
Yup

rimi-itk commented on src/Form/Type/DataFlowType.php:
```diff
@@ -37,6 +39,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'multiple' => true,
                         'expanded' => true,
                     ])
+                    ->add('schedule', TextType::class)
```
Should we use `CronExpressionType::class` here?

lats1 replied:
We could do, but it needs some customization as it's contains a lot of fields

rimi-itk commented on src/Form/Type/DataFlowType.php:
```diff
@@ -17,9 +17,11 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\Extension\Core\Type\CollectionType;
 use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
```
Is this `use` actually used?

lats1 replied:
You're right, must have sneaked in while I wasn't looking

rimi-itk commented on fixtures/geojson-change-projection.yaml:
```diff
@@ -10,7 +10,7 @@ App\Entity\DataFlow:
         name: 'GeoJSON: Change projection'
         dataSource: '@geojson_change-projection_data_source'
-        frequency: 60
+        schedule: 1 * * * *
```
Just for fun, we could let one of the `schedule` values in a single fixture be something other than `1 * * * *`.

lats1 replied:
Good idea, I will change that for some of the flows.

lats1 committed:
ODAA-154: Adding return type

lats1 committed:
ODAA-154: Cleaning up scheduling logic

lats1 committed:
ODAA-154:
- Removing unneeded use statement
- Adding info about next run time for flow to form

lats1 committed:
ODAA-154: Using different schedule values in fixtures

lats1 committed:
ODAA-154: Apply coding standards

lats1 committed:
ODAA-154: Adding translation

lats1 committed:
ODAA-154: Showing dropdown options for scheduling flows

rimi-itk reviewed (approved):
Approved!
